reposerver/repository/repository.go

```diff
@@ -1300,16 +1296,20 @@ func getObjsFromYAMLOrJson(logCtx *log.Entry, manifestPath string, filename stri
 		}
 	}()
 	if strings.HasSuffix(filename, ".json") {
-		var obj unstructured.Unstructured
+		obj := make(map[string]interface{}, 0)
 		decoder := json.NewDecoder(reader)
 		err = decoder.Decode(&obj)
 		if err != nil {
 			return status.Errorf(codes.FailedPrecondition, "Failed to unmarshal %q: %v", filename, err)
 		}
-		if decoder.More() {
-			return status.Errorf(codes.FailedPrecondition, "Found multiple objects in %q. Only single objects are allowed in JSON files.", filename)
+
+		unstructuredObj := &unstructured.Unstructured{Object: obj}
+		if unstructuredObj.GetAPIVersion() == "" ||
+			unstructuredObj.GetKind() == "" {
+			logCtx.Warnf("Skipping file, due to it not being a valid Kubernetes manifest file: %s", filename)
+			return nil
```

Comment:
The previous logic was that json can only have one object, and yaml can have multiple objects. If limit the yaml object to one, I think it might be a bug.